src/documents/cursors/common.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright DataStax, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import type { Filter, FindAndRerankCursor, FindCursor, GenericFindOptions, SomeDoc } from '@/src/documents/index.js';
+import { CursorError } from '@/src/documents/cursors/abstract-cursor.js';
+import { SerDesTarget } from '@/src/lib/api/ser-des/ctx.js';
+
+/**
+ * @internal
+ */
+export type SerializedFilter = [filter: unknown, bigNumsPresent: boolean];
+
+type FindLikeCursor = (FindCursor<any> | FindAndRerankCursor<any>);
+type FindLikeCursorConstructor<C extends FindLikeCursor> = new (...args: ConstructorParameters<typeof FindCursor | typeof FindAndRerankCursor>) => C
+
+/**
+ * @internal
+ */
+export const cloneFLC = <R, RC extends FindLikeCursor, Opts extends GenericFindOptions>(cursor: FindLikeCursor, filter: SerializedFilter, options: Opts, mapping?: (doc: SomeDoc) => R): RC => {
```

Comment:
Why is this a separate helper function instead of just a `clone()` method on AbstractCursor?